.github/workflows/ci.yml

```diff
@@ -79,11 +80,27 @@ jobs:
     if: github.event_name == 'pull_request'
     steps:
       - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # checkout all commits to be able to determine merge base
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 17
+          cache: 'maven'
+      - name: Fetch base ref to find merge-base for GIB
+        run: .github/bin/git-fetch-base-ref.sh
       - name: Check Commits
         uses: trinodb/github-actions/block-commits@c2991972560c5219d9ae5fb68c0c9d687ffcdd10
         with:
           action-merge: fail
           action-fixup: none
+      - name: Check if all commits of this PR except HEAD compile
+        run: |
+          git reset --hard HEAD~1
+          git rebase --exec "$MAVEN test $MAVEN_COMPILE_COMMITS" refs/remotes/origin/${{ github.event.pull_request.base.ref }}
```

Comment:
Why the `test` goal if there's `-DskipTests`? Would `test-compile` also be sufficient?

Reply:
Because `test` is the goal someone who runs automated bisect would be running.

Reply:
I'm skipping all tests anyway, so `test` phase would only compile tests, not run them.

Reply:
`compile test-compile` would convey the intent better